File: sdkwork-ai-sdk-swift/Sources/API/ModelsApi.swift
import Foundation

public class ModelsApi {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Retrieve model
    public func retrieveModel(model: String) async throws -> OpenAiModel? {
        let response = try await client.get(ApiPaths.aiPath("/models/\(model)"))
        return response as? OpenAiModel
    }

    /// Delete model
    public func deleteModel(model: String) async throws -> OpenAiModelDeleteResponse? {
        let response = try await client.delete(ApiPaths.aiPath("/models/\(model)"))
        return response as? OpenAiModelDeleteResponse
    }

    /// List models
    public func listModels() async throws -> OpenAiModelList? {
        let response = try await client.get(ApiPaths.aiPath("/models"))
        return response as? OpenAiModelList
    }
}

File: sdkwork-ai-sdk-kotlin/src/main/kotlin/com/sdkwork/ai/api/ModelsApi.kt
package com.sdkwork.ai.api

import com.sdkwork.ai.*
import com.sdkwork.ai.http.HttpClient

class ModelsApi(private val client: HttpClient) {

    /** Retrieve model */
    suspend fun retrieveModel(model: String): OpenAiModel? {
        return client.get(ApiPaths.aiPath("/models/$model")) as? OpenAiModel
    }

    /** Delete model */
    suspend fun deleteModel(model: String): OpenAiModelDeleteResponse? {
        return client.delete(ApiPaths.aiPath("/models/$model")) as? OpenAiModelDeleteResponse
    }

    /** List models */
    suspend fun listModels(): OpenAiModelList? {
        return client.get(ApiPaths.aiPath("/models")) as? OpenAiModelList
    }
}
